Create basic README (#2)
python-discord/forms-backend: README.md

> # Python Discord Forms (Backend)

Reviewer: ```suggestion
# Python Discord Forms API
```

> API for Python Discord forms (https://forms.pythondiscord.com) that handles Discord login and forms, submissions.

Reviewer: ```suggestion
API for [Python Discord Forms](https://forms.pythondiscord.com) that handles Discord OAuth2 integration and form submission.
```

## Setup
### Prerequisites

> To start working on forms-backend, you need few things:

Reviewer: ```suggestion
To start working on forms-backend, you'll need few things:
```

> 1. Discord Application (in [Developers Portal](https://discord.com/developers/applications))

Reviewer: ```suggestion
1. Discord OAuth2 Application (from the [Discord Developers Portal](https://discord.com/developers/applications))
```

2. Poetry
3. Docker and docker-compose (optional)
4. Running MongoDB instance (when not using Docker)

### Running with Docker

> Easiest way to run this is using Docker (and docker-compose).

Reviewer: ```suggestion
The easiest way to run forms-backend is using Docker (and docker-compose).
```

#### Environment variables

> Set following environment variables to `.env` file:

Reviewer: ```suggestion
Create a `.env` file in the root with the following values inside it:
```

> - `OAUTH2_CLIENT_ID`: Client ID of Discord Application (prerequisites point 1).
> - `OAUTH2_CLIENT_SECRET`: Client Secret of Discord Application (prerequisites point 1).

Reviewer: ```suggestion
- `OAUTH2_CLIENT_ID`: Client ID of Discord OAuth2 Application (see prerequisites).
- `OAUTH2_CLIENT_SECRET`: Client Secret of Discord OAuth2 Application (see prerequisites).
```

- `ALLOWED_URL`: Allowed origin for CORS middleware.

#### Running

> Running is really for this way: `docker-compose up`! And this is all: You can access now API from `http://localhost:8000`.

Reviewer: ```suggestion
To start using the application, simply run `docker-compose up` in the repository root. You'll be able to access the application by visiting http://localhost:8000/
```

### Running on host

> Another way to run forms-backend is manually in host.

Reviewer: ```suggestion
You can also run forms-backend manually on the host.
```

#### Environment variables

> First all same than in Docker environment variables section, with following additions:

Reviewer: ```suggestion
Create a `.env` file with the same contents as the Docker section above and the following new variables:
```

- `DATABASE_URL`: MongoDB instance URI, in format `mongodb://(username):(password)@(database IP or domain):(port)`.
- `MONGO_DB`: MongoDB database name, defaults to `pydis_forms`.

#### Running

> Almost same easy than for Docker: `uvicorn --reload --host 0.0.0.0 --debug "backend:app`.

Reviewer: ```suggestion
Simply run: `$ uvicorn --reload --host 0.0.0.0 --debug backend:app`.

Once this is running the API will live-reload on any changes.
```

# Python Discord Forms API
API for [Python Discord Forms](https://forms.pythondiscord.com) that handles Discord OAuth2 integration and form submission.

## Setup
### Prerequisites
To start working on forms-backend, you'll need few things:
1. Discord OAuth2 Application (from the [Discord Developers Portal](https://discord.com/developers/applications))
2. Poetry
3. Docker and docker-compose (optional)
4. Running MongoDB instance (when not using Docker)

### Running with Docker
The easiest way to run forms-backend is using Docker (and docker-compose).

#### Environment variables

> Create a `.env` file in the root with the following values inside it:

Reviewer: ```suggestion
Create a `.env` file in the root with the following values inside it (each variable should be a line like `VARIABLE=value`):
```

- `OAUTH2_CLIENT_ID`: Client ID of Discord OAuth2 Application (see prerequisites).
- `OAUTH2_CLIENT_SECRET`: Client Secret of Discord OAuth2 Application (see prerequisites).
- `ALLOWED_URL`: Allowed origin for CORS middleware.

#### Running
To start using the application, simply run `docker-compose up` in the repository root. You'll be able to access the application by visiting http://localhost:8000/

> ### Running on host

Reviewer: ```suggestion
### Running on the host
```

You can also run forms-backend manually on the host.

#### Environment variables
Create a `.env` file with the same contents as the Docker section above and the following new variables:
- `DATABASE_URL`: MongoDB instance URI, in format `mongodb://(username):(password)@(database IP or domain):(port)`.
- `MONGO_DB`: MongoDB database name, defaults to `pydis_forms`.

#### Running
Simply run: `$ uvicorn --reload --host 0.0.0.0 --debug backend:app`.

> Once this is running the API will live-reload on any changes.

Reviewer: We should mention what port this runs on, specify a URL like the Docker section.

I think it's:
```suggestion
Once this is running the API is accessible via http://localhost:8000/ and will reload on any changes to code.
```
